frontend/src/models/annotationsModel.ts

```diff
@@ -1,93 +1,95 @@
-                ...state,
-                {
-                    id: getNextKey(state),
-                    content,
-                    date_marker: date_marker,
-                    created_at: created_at.toISOString(),
-                    updated_at: created_at.toISOString(),
-                    created_by,
-                    scope: annotationScope || AnnotationScope.Organization,
-                } as AnnotationType,
-            ],
-            deleteGlobalAnnotation: (state, { id }) => {
-                return state.filter((a) => a.id !== id)
+            updateAnnotation: async ({
+                annotationId,
+                annotationData,
+            }: {
+                annotationId: AnnotationType['id']
+                annotationData: AnnotationDataWithoutInsight
+            }) => {
+                const updatedAnnotation = await api.annotations.update(annotationId, annotationData)
+                actions.replaceAnnotation(updatedAnnotation)
+            },
+            deleteAnnotation: async ({ annotation }) => {
+                await deleteWithUndo({
+                    endpoint: api.annotations.determineDeleteEndpoint(),
+                    object: { name: `${annotation.date_marker} annotation`, ...annotation },
+                    callback: (undo, annotation) => {
+                        if (undo) {
+                            actions.appendAnnotations([annotation])
+                        }
+                    },
+                })
+                return values.annotations.filter((a) => a.id !== annotation.id)
             },
         },
-    },
-    selectors: {
-        activeGlobalAnnotations: [
-            (s) => [s.globalAnnotations],
-            (globalAnnotations) => {
-                return globalAnnotations.filter((annotation) => !annotation.deleted)
+    })),
+    reducers(() => ({
+        annotations: [
+            [] as AnnotationType[],
+            {
+                appendAnnotations: (state, { annotations }) => [...state, ...annotations],
+                replaceAnnotation: (state, { annotation }) => {
+                    const copy = state.slice()
+                    const index = copy.findIndex((iterationAnnotation) => iterationAnnotation.id === annotation.id)
+                    copy[index] = annotation
+                    return copy
+                },
             },
         ],
-    },
-    listeners: ({ actions }) => ({
-        deleteGlobalAnnotation: ({ id }) => {
-            id >= 0 &&
-                deleteWithUndo({
-                    endpoint: `projects/${teamLogic.values.currentTeamId}/annotations`,
-                    object: { name: 'Annotation', id },
-                    callback: () => actions.loadGlobalAnnotations(),
-                })
+        next: [
+            null as string | null,
+            {
+                setNext: (_, { next }) => next,
+            },
+        ],
```

Comment:
Just a comment of my own thinking... I wonder if we could neatly encapsulate all this in a "loader" like a "paginatedLoader" that exposes the payload as well as actions for paginating. Food for thought for the future.
I've seen this implemented enough times for it to probably make some sense

Reply:
Yeah, a loader pattern with `nextUrl` would be great. Also, it'd be awesome to integrate its UX into `LemonTable` – currently those "Load more" buttons are weirdly outside of their tables.